libtransmission/rpcimpl.c

```diff
@@ -927,6 +940,29 @@ static char const* torrentGet(tr_session* session, tr_variant* args_in, tr_varia
 ****
 ***/
 
+static char const* setLabels(tr_torrent* tor, tr_variant* list)
+{
+    const char * str;
+    size_t str_len;
+    const int n = tr_variantListSize(list);
+    for (int i = 0; i < n; i++)
+    {
+        if (tr_variantGetStr(tr_variantListChild(list, i), &str, &str_len) && str && str_len)
+        {
+            if (strchr(str, ',') != NULL)
+                return "labels cannot contain comma (,) character";
+        }
+    }
+    tr_ptrArrayDestruct(&tor->labels, tr_free);
+    tor->labels = TR_PTR_ARRAY_INIT;
+    for (int i = 0; i < n; i++)
+    {
+        if (tr_variantGetStr(tr_variantListChild(list, i), &str, &str_len) && str && str_len)
+            tr_ptrArrayAppend(&tor->labels, tr_strndup(str, str_len));
+    }
+    return NULL;
```

Comment:
Free `list` here, could (and in current code will) result in memory leak if variable is reused since next call to `tr_variantDictFindList` (in `torrentSet` below) doesn't care.